service/matching/matching_engine.go

```diff
@@ -948,8 +948,24 @@ func (e *matchingEngineImpl) DescribeTaskQueue(
 					if physInfo, ok := physicalInfoByBuildId[buildId][taskQueueType]; !ok {
 						physicalInfoByBuildId[buildId][taskQueueType] = vii.PhysicalTaskQueueInfo
 					} else {
+						var mergedStats *taskqueuepb.TaskQueueStats
+
+						// only report Task Queue Statistics if requested.
+						if req.GetReportStats() {
+							rootStats := physicalInfoByBuildId[buildId][taskQueueType].TaskQueueStats // BacklogInfo of the previous partition
```

Comment:
Shall be renaming this to `previousPartitionStats` and `currentPartitionStats` in the next push. Thanks @carlydf for the suggestion!

Reply:
I missed this.. what is "previous" and "current" here?

Reply:
"previous" would refer to the stats collected from the previous partitions of the tq while current refers to the stats returned for the given partition

I suggested that renaming them would be better since it would make things easier for a reader